sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/test/TestTableProvider.java

```diff
@@ -150,12 +165,40 @@ public BeamTableStatistics getTableStatistics(PipelineOptions options) {
       return begin.apply(Create.of(tableWithRows.rows).withCoder(rowCoder()));
     }
 
+    @Override
+    public PCollection<Row> buildIOReader(
+        PBegin begin, BeamSqlTableFilter filters, List<String> fieldNames) {
+      PCollection<Row> withAllFields = buildIOReader(begin);
+      if (options == PushDownOptions.NONE
+          || (fieldNames.isEmpty() && filters instanceof DefaultTableFilter)) {
+        return withAllFields;
+      }
```

Comment:
Some portion of this `if` statement can be moved to `BeamIOSourceRel`, more specifically the `Transform#expand` method.

Reply:
Moved the default case to `Transform#expand`.